js/buda.js

```diff
@@ -369,7 +369,7 @@ module.exports = class buda extends Exchange {
         const open = parseFloat (this.priceToPrecision (symbol, last / (percentage + 1)));
         const change = last - open;
         const average = this.sum (last, open) / 2;
-        return {
+        return this.safeTicker ({
```

Comment:
We should remove lines 369-371 (the `open`, `change`, `average` calculation), cause it's done in safeTicker. We should set them to `undefined` in `parseTicker`, if they're not explicitly defined in the exchange response.
We should also replace lines 363-366:
```JavaScript
const symbol = this.safeSymbol (undefined, market);
```

Reply:
Actually, if you check their verbose response:

```
node examples/js/cli buda fetchTicker ETH/BTC
```

They will return a response like this:

```
{"ticker":{"market_id":"ETH-BTC","last_price":["0.07300001","BTC"],"min_ask":["0.07716895","BTC"],"max_bid":["0.0754966","BTC"],"volume":["0.168965697","ETH"],"price_variation_24h":"-0.046","price_variation_7d":"-0.085"}}
```

And their response contains a `market_id`. We could parse that in `parseTicker` to determine the `symbol` if it's available. We should prioritize the market info from inside the ticker itself over the market info coming from the outside parseTicker.

It would help a lot if you could also add verbose response samples in fetch-methods and in parse-methods in the following way:

- in `fetch` methods immediately after `await` line:

```JavaScript
    async fetchTicker (symbol, params = {}) {
        await this.loadMarkets ();
        const market = this.market (symbol);
        const request = {
            'market': market['id'],
        };
        const response = await this.publicGetMarketsMarketTicker (this.extend (request, params));
        //
        //     {
        //         "ticker":{
        //             "market_id":"ETH-BTC",
        //             "last_price":["0.07300001","BTC"],
        //             "min_ask":["0.07716895","BTC"],
        //             "max_bid":["0.0754966","BTC"],
        //             "volume":["0.168965697","ETH"],
        //             "price_variation_24h":"-0.046",
        //             "price_variation_7d":"-0.085"
        //         }
        //     }
        //
        const ticker = this.safeValue (response, 'ticker');
        return this.parseTicker (ticker, market);
    }
```

- in `parse` methods at the top, to show the structure that is being parsed:

```JavaScript
    parseTicker (ticker, market = undefined) {
        //
        // fetchTicker
        //
        //     {
        //         "market_id":"ETH-BTC",
        //         "last_price":["0.07300001","BTC"],
        //         "min_ask":["0.07716895","BTC"],
        //         "max_bid":["0.0754966","BTC"],
        //         "volume":["0.168965697","ETH"],
        //         "price_variation_24h":"-0.046",
        //         "price_variation_7d":"-0.085"
        //     }
        //
        const timestamp = this.milliseconds ();
        const marketId = this.safeString (ticker, 'market_id');
        const symbol = this.safeSymbol (marketId, market, '-');
        const lastPrice = this.safeValue (ticker, 'last_price', []);
        const last = this.safeNumber (lastPrice, 0);
        const percentage = this.safeNumber (ticker, 'price_variation_24h');
        return this.safeTicker ({
        // ...
```

As you can see from the above structure, we could also fill-in the infos from `min_ask` and `max_bid`.

Reply:
I probably should have let someone else adjust this exchange, if I call this method using
```
node examples/js/cli buda fetchTicker ETH/BTC --verbose
```
I get this error:
```
[ExchangeNotAvailable] buda GET https://www.buda.com/api/v2/currencies 503 Service Temporarily Unavailable
```
I made a few more recommended changes but wasn't able to test them

Reply:
Does this URL work for you in the browser: https://www.buda.com/api/v2/currencies ? If not, no worries, I will finish and will merge.

Reply:
Yeah that URL works for me in the browser, I'm not sure why I'm getting a 503 error calling it from the command line. I think the changes I made should be good